src/modules/launcher/Wox.Infrastructure/Storage/JsonStorage`1.cs

```diff
@@ -122,18 +126,146 @@ private void BackupOriginFile()
 
         public void Save()
         {
+            lock (_saveLock)
+            {
+                try
+                {
+                    string serialized = JsonSerializer.Serialize(_data, _serializerOptions);
+                    File.WriteAllText(FilePath, serialized);
+
+                    Log.Info($"Saving cached data at <{FilePath}>", GetType());
+                }
+                catch (IOException e)
+                {
+                    Log.Exception($"Error in saving data at <{FilePath}>", e, GetType());
+                }
+            }
+        }
+
+        public void Clear()
+        {
+            if (File.Exists(FilePath))
+            {
+                File.Delete(FilePath);
+                LoadDefault();
+                Log.Info($"Deleting cached data at <{FilePath}>", GetType());
+            }
+        }
+
+        public void SaveInformationFile(T data)
+        {
+            lock (_saveLock)
+            {
+                DefaultFileContent = JsonSerializer.Serialize(data, _informationSerializerOptions);
+                _storageHelper.Close(DefaultFileContent);
+            }
+        }
+
+        public bool CheckVersionMismatch(T actualData, bool versionMismatch = false)
+        {
+            // Skip the fields check if the version hasn't changed.
+            // This optimization prevents unnecessary fields processing when the cache
+            // is already up to date, enhancing performance and reducing IO operations
+            if (!_storageHelper.ClearCache && !versionMismatch)
+            {
+                DefaultFileContent = JsonSerializer.Serialize(actualData, _informationSerializerOptions);
+                _storageHelper.Close(DefaultFileContent);
+                return false;
+            }
+
+            return true;
+        }
+
+        public bool CheckWithInformationFileToClear(T actualData)
+        {
+            var infoFilePath = _storageHelper.GetFilePath(FilePath, _jsonStorage);
+
+            if (actualData == null)
+            {
+                return false;
+            }
+            else if (!File.Exists(infoFilePath))
+            {
+                // Check if information file exist
+                DefaultFileContent = JsonSerializer.Serialize(actualData, _informationSerializerOptions);
+
+                _storageHelper.Close(DefaultFileContent);
+                return true;
+            }
+
             try
             {
-                string serialized = JsonSerializer.Serialize(_data, _serializerOptions);
-                File.WriteAllText(FilePath, serialized);
-                _storageHelper.Close();
+                var infoFields = JsonSerializer.Deserialize<Dictionary<string, object>>(File.ReadAllText(infoFilePath), _informationSerializerOptions);
```

Comment:
I can't figure out why this entire logic is needed? i.e. why can we do something like:
```
try deserialize the file as it is
if it fails, it's either malformed or have new fields. in that case override it
```

could you elaborate on why this wouldn't work?